packages/ember-glimmer/lib/environment.js

```diff
@@ -257,12 +257,16 @@ export default class Environment extends GlimmerEnvironment {
     }
   }
 
-  hasHelper(name) {
-    return !!builtInHelpers[name[0]] || this.owner.hasRegistration(`helper:${name}`);
+  hasHelper(name, parentMeta) {
+    return !!builtInHelpers[name[0]] ||
+      this.owner.hasRegistration(`helper:${name}`) ||
+      (parentMeta && this.owner.hasRegistration(`helper:${name}`, { source: `template:${parentMeta.moduleName}` }));
   }
 
-  lookupHelper(name) {
-    let helper = builtInHelpers[name[0]] || this.owner.lookup(`helper:${name}`);
+  lookupHelper(name, parentMeta) {
+    let helper = builtInHelpers[name[0]] ||
+      this.owner.lookup(`helper:${name}`) ||
+      (parentMeta && this.owner.lookup(`helper:${name}`, { source: `template:${parentMeta.moduleName}` }));
```

Comment:
@krisselden @rwjblue for some reason, doing a `lookup` on a helper with `source='-top-level'` doesn't work. Same for `hasRegistration`. This works for components, but not helpers. To demonstrate, I've added a `lookup` without the `source` info, and then a fallback with `source` info, and things work. Any idea why this is?